frostsnapp/native/src/wallet.rs

```diff
@@ -212,18 +216,16 @@ impl _Wallet {
         // reveal_next_spk will panic if the key doesn't exist
         self.ensure_key_exists(key_id)?;
```

Comment:
Is there a case where this is called before creating a key? is this a pattern we should follow elsewhere (checking inputs are valid)?

Reply:
I tried to clear up this call to make it more self evident: e102d341cd3cac18f50c20afd93166c9d54cc359

In general you should avoid designing methods that check inputs and valid (and that's not what this method does). You try and make APIs correct by construction by defining types with no invalid instances. This allows the caller to try and convert their data to the type and deal with the errors there before it reaches your code.